### ISSUE TITLE
Export-NavContainerDatabasesAsBacpac error deleting [user login] data

### ISSUE THREAD

**Issue by ekeukens** [Microsoft/navcontainerhelper#421]
Just updated to container helper 0.6.0.3. I created new-navcontainer, used same source bakfile as I did before. Export to bacpac cmdlet comes with an error (see below). 
I was not aware of data in these tables. My previous version of navcontainer helper did not bother (or did delete). 
I deleted data in [user login] tables, created new source bak file (and navcontainer). Still an error when exporting. Because I know tables are empty, I skipped Invoke-Sqlcmd for deleting this data. The script runs OK.

Successfully exported database and saved it to file 'C:\ProgramData\NavContainerHelper\Extensions\dev021nlmt\app.bacpac'.
Copying Database on localhost\SQLEXPRESS from tenant to tempTenant
Removing Database tempTenant from localhost\SQLEXPRESS
Taking database tenant offline
Copying database files
Attaching files as new Database tempTenant
Putting database tenant back online
### _**Remove data from User table and related tables in tempTenant database.
DELETE FROM dbo.[Contoso Demo Install bedrijf$User Login]
The WriteObject and WriteError methods cannot be called from outside the overrides of the BeginProcessing, ProcessRecord, and EndProcessing methods, and they can only be called from within the same 
thread. Validate that the cmdlet makes these calls correctly, or contact Microsoft Customer Support Services.
at Remove-NavTenantDatabaseUserData, <No file>: line 124
at <ScriptBlock>, <No file>: line 220
at <ScriptBlock>, <No file>: line 216**_

**Comment by freddydk** [comment]
I can see this behavior was added in 0.5.0.2. I assume you were running an older version before that?
I cannot repro the scenario - what database are you using? (country?)
Are you running as admin?

**Comment by freddydk** [comment]
Do you have multiple companies in the database?

**Comment by ekeukens** [comment]
Yes 2.

Vriendelijke groet | Kind regards,




Edwin Keukens
Migratie- en release coördinator NAV

+31 6 51253699<tel:+31%206%2051253699>
www.4ps.nl<https://www.4ps.nl>

[https://www.4ps.nl/userfiles/image/logo(1).png]

[https://www.4ps.nl/userfiles/image/footer.png]


Van: Freddy Kristiansen [mailto:notifications@github.com]
Verzonden: donderdag 11 april 2019 23:57
Aan: Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com>
CC: Edwin Keukens <ekeukens@4PS.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [Microsoft/navcontainerhelper] Export-NavContainerDatabasesAsBacpac error deleting [user login] data (#421)


Do you have multiple companies in the database?

**Comment by ekeukens** [comment]
Hello,

I’ve created container with imagename = "microsoft/bcsandbox:nl"

So NL database.

NavContainerHelper is version 0.6.0.3
Host is Microsoft Windows 10 Pro - 1803
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Using image microsoft/bcsandbox:nl-ltsc2016
Removing container dev021nl
Removing dev021nl from hosts
Removing C:\ProgramData\NavContainerHelper\Extensions\dev021nl
Creating Nav container dev021nl
Using license file C:\MergeToBC\bacpac\4805272.flf
Version: 13.4.28874.29517-NL
Platform: 13.0.28871.29362
Generic Tag: 0.0.9.2
Container OS Version: 10.0.14393.2791 (ltsc2016)
Host OS Version: 10.0.17134.706 (1803)
The container operating system does not match the host operating system, forcing hyperv isolation.
Using hyperv isolation
Creating container dev021nl from image microsoft/bcsandbox:nl-ltsc2016



Vriendelijke groet | Kind regards,




Edwin Keukens
Migratie- en release coördinator NAV

+31 6 51253699<tel:+31%206%2051253699>
www.4ps.nl<https://www.4ps.nl>

[https://www.4ps.nl/userfiles/image/logo(1).png]

[https://www.4ps.nl/userfiles/image/footer.png]


Van: Freddy Kristiansen [mailto:notifications@github.com]
Verzonden: donderdag 11 april 2019 23:51
Aan: Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com>
CC: Edwin Keukens <ekeukens@4PS.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [Microsoft/navcontainerhelper] Export-NavContainerDatabasesAsBacpac error deleting [user login] data (#421)


I can see this behavior was added in 0.5.0.2. I assume you were running an older version before that?
I cannot repro the scenario - what database are you using? (country?)
Are you running as admin?

**Comment by ekeukens** [comment]
I assume I can continue uploading this bacpac to embed app in LCS?
I only switched off deleting this data, but I checked that in both companies this table is empty. So the created bacpac (ran ok after switching off this statement) must be OK.
In the script I see this is the only user table which is company specific.

Vriendelijke groet | Kind regards,




Edwin Keukens
Migratie- en release coördinator NAV

+31 6 51253699<tel:+31%206%2051253699>
www.4ps.nl<https://www.4ps.nl>

[https://www.4ps.nl/userfiles/image/logo(1).png]

[https://www.4ps.nl/userfiles/image/footer.png]


Van: Edwin Keukens
Verzonden: donderdag 11 april 2019 23:58
Aan: 'Microsoft/navcontainerhelper' <reply@reply.github.com>; Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com>
CC: Author <author@noreply.github.com>
Onderwerp: RE: [Microsoft/navcontainerhelper] Export-NavContainerDatabasesAsBacpac error deleting [user login] data (#421)

Hello,

I’ve created container with imagename = "microsoft/bcsandbox:nl"

So NL database.

NavContainerHelper is version 0.6.0.3
Host is Microsoft Windows 10 Pro - 1803
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Using image microsoft/bcsandbox:nl-ltsc2016
Removing container dev021nl
Removing dev021nl from hosts
Removing C:\ProgramData\NavContainerHelper\Extensions\dev021nl
Creating Nav container dev021nl
Using license file C:\MergeToBC\bacpac\4805272.flf
Version: 13.4.28874.29517-NL
Platform: 13.0.28871.29362
Generic Tag: 0.0.9.2
Container OS Version: 10.0.14393.2791 (ltsc2016)
Host OS Version: 10.0.17134.706 (1803)
The container operating system does not match the host operating system, forcing hyperv isolation.
Using hyperv isolation
Creating container dev021nl from image microsoft/bcsandbox:nl-ltsc2016



Vriendelijke groet | Kind regards,





Edwin Keukens
Migratie- en release coördinator NAV

+31 6 51253699<tel:+31%206%2051253699>
www.4ps.nl<https://www.4ps.nl>

[https://www.4ps.nl/userfiles/image/logo(1).png]

[https://www.4ps.nl/userfiles/image/footer.png]


Van: Freddy Kristiansen [mailto:notifications@github.com]
Verzonden: donderdag 11 april 2019 23:51
Aan: Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com<mailto:navcontainerhelper@noreply.github.com>>
CC: Edwin Keukens <ekeukens@4PS.nl<mailto:ekeukens@4PS.nl>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Onderwerp: Re: [Microsoft/navcontainerhelper] Export-NavContainerDatabasesAsBacpac error deleting [user login] data (#421)


I can see this behavior was added in 0.5.0.2. I assume you were running an older version before that?
I cannot repro the scenario - what database are you using? (country?)
Are you running as admin?

**Comment by freddydk** [comment]
Yes, you should be good

**Comment by freddydk** [comment]
The right fix for this is
```
                $tables = Invoke-Sqlcmd @params -Query "USE [$DatabaseName] SELECT name FROM sysobjects WHERE (xtype = 'U' ) AND (name LIKE '%User Login')"
                $tables | % {
                    Write-Host "DELETE FROM dbo.[$($_.Name)]"
                    Invoke-Sqlcmd @params -Query "USE [$DatabaseName] DELETE FROM dbo.[$($_.Name)]"
                }
```
To end the first Invoke-SqlCmd before running into the next.
Will be fixed in the next update - probably later tonight.

**Comment by ekeukens** [comment]
Thanks for quick response Freddy, really appreciate..
Almost up and running with embed app at 4ps.. Thanks again..

Vriendelijke groet | Kind regards,




Edwin Keukens
Migratie- en release coördinator NAV

+31 6 51253699<tel:+31%206%2051253699>
www.4ps.nl<https://www.4ps.nl>

[https://www.4ps.nl/userfiles/image/logo(1).png]

[https://www.4ps.nl/userfiles/image/footer.png]


Van: Freddy Kristiansen [mailto:notifications@github.com]
Verzonden: vrijdag 12 april 2019 00:08
Aan: Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com>
CC: Edwin Keukens <ekeukens@4PS.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [Microsoft/navcontainerhelper] Export-NavContainerDatabasesAsBacpac error deleting [user login] data (#421)


The right fix for this is

                $tables = Invoke-Sqlcmd @params -Query "USE [$DatabaseName] SELECT name FROM sysobjects WHERE (xtype = 'U' ) AND (name LIKE '%User Login')"

                $tables | % {

                    Write-Host "DELETE FROM dbo.[$($_.Name)]"

                    Invoke-Sqlcmd @params -Query "USE [$DatabaseName] DELETE FROM dbo.[$($_.Name)]"

                }

To end the first Invoke-SqlCmd before running into the next.
Will be fixed in the next update - probably later tonight.

**Comment by freddydk** [comment]
Shipped in 0.6.0.10
